ephios/core/views/pwa.py

```diff
@@ -48,7 +48,17 @@ def get_context_data(self, **kwargs):
         context["offline_url"] = reverse("core:pwa_offline")
         # Cache name: we serve /static/ files with a cache first strategy, so
         # we need to use a new cache name when the static files change with a new ephios version.
-        context["cache_name"] = f"ephios-pwa-v{settings.EPHIOS_VERSION}"
+        # Also, we need to start a new cache for every user and permission change.
+        identity = self.request.user.id if self.request.user.is_authenticated else "anonymous"
+        permissions = set(self.request.user.get_all_permissions())
```

Comment:
could this be a performance issue? if I understand it correctly, this needs a fresh DB query for every request?

Reply:
yes, but the serviceworker is loaded asynchronously, so it does not affect the page loads.